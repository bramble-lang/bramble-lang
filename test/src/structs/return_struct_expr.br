fn my_main() -> i64 {
    // Test to make sure that empty structs can be returned and do not mess up anything
    let e: EmptyStruct := return_empty();

    // Test that struct expressions can be passed and returned
    // Pass two to test if multiple parameters cause corruption 
    let y: MyStruct := test4(MyStruct{ x: 5, y: 75, z: 18}, MyStruct{ x: 15, y: 125, z: 22});
    root::std::io::writei64ln(y.x);
    root::std::io::writei64ln(y.y);
    root::std::io::writei64ln(y.z);

    
    return 0;
}

struct MyStruct {
    x: i64,
    y: i64,
    z: i64,
}

fn test4(x: MyStruct, y: MyStruct) -> MyStruct {
    return MyStruct{
        x: x.x + y.x,
        y: x.y + y.y,
        z: x.z + y.z,
    };
}

struct EmptyStruct {
}

fn return_empty() -> EmptyStruct {
    return EmptyStruct{};
}