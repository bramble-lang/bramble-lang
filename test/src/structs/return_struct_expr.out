10
100
20
